src/develop/masks/path.c

```diff
@@ -354,10 +354,10 @@ static size_t _path_find_self_intersection(dt_masks_dynbuf_t *inter,
   size_t inter_count = 0;
 
   // we search extreme points in x and y
-  int xmin, xmax, ymin, ymax;
+  size_t xmin, xmax, ymin, ymax;
   xmin = ymin = INT_MAX;
-  xmax = ymax = INT_MIN;
-  int posextr[4] = { -1 }; // xmin,xmax,ymin,ymax
+  xmax = ymax = 0;
+  size_t posextr[4] = { 0 }; // xmin,xmax,ymin,ymax
```

Comment:
are you sure this is safe? replacing -1 by 0? 0 sounds like a valid number, -1 not so much..

Reply:
I did not see that there a < 0 check in the code for those values. If such a check is missing then I might be wrong ...